### PR TITLE
Update mblock-mlink sha256 hash

### PR DESCRIPTION
**Pull request by levigross**
## Description of changes

Update mblock-mlink sha256 hash
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by levigross** [comment]
```$ nix-shell -p mblock-mlink
these 2 derivations will be built:
  /nix/store/wxxymjr2xad5py8j3ijf0ck4kzpvird4-mLink-1.2.0-amd64.deb.drv
  /nix/store/4vg70myznsp33aqx8ljfcflnl1np4d80-mblock-mlink-1.2.0.drv
these 21 paths will be fetched (5.54 MiB download, 24.92 MiB unpacked):
  /nix/store/jxf3a0wbn5xf1fn191j6ij6ya92z6jaw-auto-patchelf-hook
  /nix/store/4wzbq7sbi9pvadlv0gkx73rya3snajq4-binutils-wrapper-2.41
  /nix/store/wpm6r5zrzy550adld2k0saji3f7a4r3a-curl-8.7.1
  /nix/store/xim3bgyqa53janqhb4fyvrfjlp6gqfqc-curl-8.7.1-bin
  /nix/store/pm49xg09dwhj6kfsahfdmz0w05il7fq7-curl-8.7.1-dev
  /nix/store/0hqh71yvg6w2n926fyn014jc7k6y4ks3-curl-8.7.1-man
  /nix/store/mgzpz6ga6y2i9c26w1v63l5xnzbavbyd-dpkg-1.22.5
  /nix/store/8k0wzaz4n1g0jalm2grxxmm3cx5qnr69-libkrb5-1.21.3
  /nix/store/7apqaj019kf1ikqx4sbcmaniiv7c9wx2-libkrb5-1.21.3-dev
  /nix/store/shvfy2h0s9ibfnfxglykzhc10cwlnjs7-libssh2-1.11.0
  /nix/store/75whk8wxyyfvvbzaxmss60zya17q5g7w-libssh2-1.11.0-dev
  /nix/store/bvwhr8x5bh0hcbirr3vwb7yq5vn2kb98-mirrors-list
  /nix/store/w4viw5mzdkz1p4cqdsj2p4d860q2i6b8-nghttp2-1.61.0
  /nix/store/fm945illxnl15i679gsaa8q9h7xlpi4x-nghttp2-1.61.0-dev
  /nix/store/zarjmh0hj3lgy3fgfqcvv47lhf1g315p-nghttp2-1.61.0-lib
  /nix/store/i4s93chycf44s3824m83fw4ssj07fv5h-openssl-3.0.14
  /nix/store/1kvd545c3qa1sq68qrrnbwazghd5wxxr-openssl-3.0.14-bin
  /nix/store/qpqcaxdv6a18bz5fk60rpkvpbsba12jz-openssl-3.0.14-dev
  /nix/store/xrfa7pgb5qv42xslnc07ah3zh4g9wc8h-python3-3.11.9-env
  /nix/store/b0m9x816h8mzr4bnhkbqw2knbn18ilyq-python3.11-pyelftools-0.31
  /nix/store/yywfkjbmf0czk408qrqm691kclffvmg0-stdenv-linux
copying path '/nix/store/mgzpz6ga6y2i9c26w1v63l5xnzbavbyd-dpkg-1.22.5' from 'https://cache.nixos.org'...
copying path '/nix/store/4wzbq7sbi9pvadlv0gkx73rya3snajq4-binutils-wrapper-2.41' from 'https://cache.nixos.org'...
copying path '/nix/store/bvwhr8x5bh0hcbirr3vwb7yq5vn2kb98-mirrors-list' from 'https://cache.nixos.org'...
copying path '/nix/store/0hqh71yvg6w2n926fyn014jc7k6y4ks3-curl-8.7.1-man' from 'https://cache.nixos.org'...
copying path '/nix/store/b0m9x816h8mzr4bnhkbqw2knbn18ilyq-python3.11-pyelftools-0.31' from 'https://cache.nixos.org'...
copying path '/nix/store/w4viw5mzdkz1p4cqdsj2p4d860q2i6b8-nghttp2-1.61.0' from 'https://cache.nixos.org'...
copying path '/nix/store/yywfkjbmf0czk408qrqm691kclffvmg0-stdenv-linux' from 'https://cache.nixos.org'...
copying path '/nix/store/8k0wzaz4n1g0jalm2grxxmm3cx5qnr69-libkrb5-1.21.3' from 'https://cache.nixos.org'...
copying path '/nix/store/zarjmh0hj3lgy3fgfqcvv47lhf1g315p-nghttp2-1.61.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/i4s93chycf44s3824m83fw4ssj07fv5h-openssl-3.0.14' from 'https://cache.nixos.org'...
copying path '/nix/store/fm945illxnl15i679gsaa8q9h7xlpi4x-nghttp2-1.61.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/xrfa7pgb5qv42xslnc07ah3zh4g9wc8h-python3-3.11.9-env' from 'https://cache.nixos.org'...
copying path '/nix/store/7apqaj019kf1ikqx4sbcmaniiv7c9wx2-libkrb5-1.21.3-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/jxf3a0wbn5xf1fn191j6ij6ya92z6jaw-auto-patchelf-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/shvfy2h0s9ibfnfxglykzhc10cwlnjs7-libssh2-1.11.0' from 'https://cache.nixos.org'...
copying path '/nix/store/1kvd545c3qa1sq68qrrnbwazghd5wxxr-openssl-3.0.14-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/wpm6r5zrzy550adld2k0saji3f7a4r3a-curl-8.7.1' from 'https://cache.nixos.org'...
copying path '/nix/store/qpqcaxdv6a18bz5fk60rpkvpbsba12jz-openssl-3.0.14-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/xim3bgyqa53janqhb4fyvrfjlp6gqfqc-curl-8.7.1-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/75whk8wxyyfvvbzaxmss60zya17q5g7w-libssh2-1.11.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/pm49xg09dwhj6kfsahfdmz0w05il7fq7-curl-8.7.1-dev' from 'https://cache.nixos.org'...
building '/nix/store/wxxymjr2xad5py8j3ijf0ck4kzpvird4-mLink-1.2.0-amd64.deb.drv'...

trying https://dl.makeblock.com/mblock5/linux/mLink-1.2.0-amd64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   242  100   242    0     0    315      0 --:--:-- --:--:-- --:--:--   315
100  5650    0  5650    0     0   3404      0 --:--:--  0:00:01 --:--:--  6823
error: hash mismatch in fixed-output derivation '/nix/store/wxxymjr2xad5py8j3ijf0ck4kzpvird4-mLink-1.2.0-amd64.deb.drv':
         specified: sha256-KLxj81ZjbEvhhaz0seNB4WXX5ybeZ7/WcT1dGfdWle0=
            got:    sha256-vwIzotvplkE8gjQe25bEqTF/3N3WK7weqGmbK/HAyVA=
error: 1 dependencies of derivation '/nix/store/4vg70myznsp33aqx8ljfcflnl1np4d80-mblock-mlink-1.2.0.drv' failed to build
```

File: pkgs/development/tools/mblock-mlink/default.nix [hunk 1]
```diff
@@ -6,7 +6,7 @@ stdenv.mkDerivation rec {
 
   src = fetchurl {
     url = "https://dl.makeblock.com/mblock5/linux/mLink-${version}-amd64.deb";
-    sha256 = "sha256-KLxj81ZjbEvhhaz0seNB4WXX5ybeZ7/WcT1dGfdWle0=";
+    sha256 = "sha256-vwIzotvplkE8gjQe25bEqTF/3N3WK7weqGmbK/HAyVA=";
```
Comment (drupol) [inline pkgs/development/tools/mblock-mlink/default.nix:9]:
```suggestion
    hash = "sha256-vwIzotvplkE8gjQe25bEqTF/3N3WK7weqGmbK/HAyVA=";
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — CHANGES_REQUESTED** [review]
Thanks!

1) Make sure the PR title is valid.
2) Make sure the commit log message is valid.
3) Make sure there's only one commit in this PR.

**Comment by mausch** [comment]
This is bizarre... they've changed the file? Also for some reason it's not getting recognised as a Debian package :confused: 

```
$ NIXPKGS_ALLOW_UNFREE=1 nix run -L --impure github:nixos/nixpkgs/70c93acc04ca47e42d06bdff914261b8bcabc951#mblock-mlink
mblock-mlink> Running phase: unpackPhase
mblock-mlink> dpkg-deb: error: '/nix/store/s5ywm3q4s989mvzy6vpm6faks4jd466w-mLink-1.2.0-amd64.deb' is not a Debian format archive
error: builder for '/nix/store/n6an99rrqgdnzz80gqhalkvyl46z4svg-mblock-mlink-1.2.0.drv' failed with exit code 2;
       last 2 log lines:
       > Running phase: unpackPhase
       > dpkg-deb: error: '/nix/store/s5ywm3q4s989mvzy6vpm6faks4jd466w-mLink-1.2.0-amd64.deb' is not a Debian format archive
       For full logs, run 'nix log /nix/store/n6an99rrqgdnzz80gqhalkvyl46z4svg-mblock-mlink-1.2.0.drv'.
```

**Pull request closed without merging**